### ISSUE TITLE
[Feature Request] Allow jei within quest book

### ISSUE THREAD

**Issue by Epb7304** [lorddusk/HQM#691]
**TL/DR:**
Allow players who have jei (or a similar mod installed) to check the recipe of an item while in the quest book

**The issue to be resolved:**
Currently, and especially in larger mod packs it can be tough to remember the list of items that you may need to acquire for a quest and to type them all into jei

**Proposed Solution:**
allow jei and similar mods to be used while using the quest book, and when backing out of the jei interface, bring them back to the book.
so if I wanted to check the recipe of an item I need for a quest I could just hit `R` instead of backing out of the book, remembering the potentially 5+ items needed, and typing them into Jei's search bar.
